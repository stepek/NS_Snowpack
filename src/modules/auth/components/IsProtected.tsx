import React, {ReactNode} from "react"
import {useSelector} from "react-redux"

import {getAuthorizedState} from "../authDuck/selectors"

import {Login} from "./Login"

interface IsProtectedProps {
  children?: ReactNode
}
export function IsProtected({children}: IsProtectedProps) {
  const isAuthorized = useSelector(getAuthorizedState)

  if (isAuthorized) {
    return <>{children}</> // Make ts happy
  }

  return <Login />
}

export default IsProtected
