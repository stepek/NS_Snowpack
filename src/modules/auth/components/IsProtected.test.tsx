import React from "react"
import {useSelector} from "react-redux"
import renderer from "react-test-renderer"

import IsProtected from "./IsProtected"

jest.mock("./Login", () => ({
  Login: () => <div>LoginComponent</div>,
}))

jest.mock("react-redux")

afterEach(() => {
  useSelector.mockClear()
})

test("<IsProtected> render correctly for non authorized", () => {
  useSelector.mockImplementationOnce(() => false)

  expect(renderer.create(<IsProtected />).toJSON()).toMatchSnapshot()
})

test("<IsProtected> render correctly for authorized component", () => {
  useSelector.mockImplementationOnce(() => true)

  expect(
    renderer
      .create(
        <IsProtected>
          <div>Hidden component</div>
        </IsProtected>,
      )
      .toJSON(),
  ).toMatchSnapshot()
})
