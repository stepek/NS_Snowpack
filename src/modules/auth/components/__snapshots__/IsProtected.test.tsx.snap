// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IsProtected> render correctly for authorized component 1`] = `
<div>
  Hidden component
</div>
`;

exports[`<IsProtected> render correctly for non authorized 1`] = `
<div>
  LoginComponent
</div>
`;
